src/tools/pkcs15-crypt.c

```diff
@@ -143,7 +144,8 @@ static char * get_pin(struct sc_pkcs15_object *obj)
 			return strdup(opt_pincode);
 	}
 
-	sprintf(buf, "Enter PIN [%.*s]: ", (int) sizeof obj->label, obj->label);
+	snprintf(buf, sizeof(buf), "Enter PIN [%.*s]: ",
+		(int) MIN(strlen(obj->label), sizeof(buf) - 15), obj->label);
```

Comment:
same here, you could also change the size of `buf`

Reply:
True. It will make it simpler, but the `snprintf()` should be used anyway.